### ISSUE TITLE
Unable to undo (cntrl-z or undo button) after deleting

### ISSUE THREAD

**Issue by bryantwong** [sagemath/cloud#118]
In a .sagews document, I'm able to replicate an issue where undoing doesn't work.

If I start in some cell above the markup that gives a problem (see attached image), then press Shift + Down-Arrow some number of times until I've highlighted the markup and some cells above and below that
![screenshot from 2014-04-03 23 52 58](https://cloud.githubusercontent.com/assets/7159227/2612570/d7460f4e-bbc5-11e3-9e94-880746427150.png), 

and press delete, 

![screenshot from 2014-04-03 23 53 09](https://cloud.githubusercontent.com/assets/7159227/2612578/16ea5740-bbc6-11e3-9b56-8d21deeb673e.png)

I am unable to undo the deletion. 

![screenshot from 2014-04-03 23 54 43](https://cloud.githubusercontent.com/assets/7159227/2612575/0f2a9830-bbc6-11e3-85bd-2fe04ab57ae1.png)

Neither control/command-z or the undo button will return the document back to its previous state.

**Comment by haraldschilly** [comment]
might be a silly question, so late, but do you still have that worksheet? send us a link at help@sagemath.com with its URL.

**Comment by williamstein** [comment]
I followed the directions above (with the screenshots) and was very easily able to replicate exactly this error.   I just started with any very simple worksheet.   It's a serious problem that we'll definitely want to fix.

**Comment by bryantwong** [comment]
I don't think this problem is worksheet-specific (I accidentally experienced this problem many times in my class), but I believe the specific worksheet is located here: 

https://cloud.sagemath.com/projects/741fb650-cb46-469a-be47-befe9db6ed8a/files/homework1/hw1.sagews

I'm not sure if you will be able to access it but I can add you as collaborators as necessary.
